### PR TITLE
Galal hussein storage class tasks

### PR DESCRIPTION
**Pull request by MBishop17**
Initial draft of storage documentation

### CODE REVIEW

File: content/rancher/v2.x/en/concepts/global-configuration/users-permissions-roles/_index.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
+        - [Cluster Roles](#cluster-roles)
+            - [Custom Cluster Roles](#custom-cluster-roles)
+            - [Cluster Role Reference](#cluster-role-reference)
+        - [Project Roles](#project-roles)
+            - [Custom Project Roles](#custom-project-roles)
+            - [Project Role Reference](#project-role-reference)
+    - [Defining Custom Roles](#defining-custom-roles)
+        - [Locked Roles](#locked-roles)
+
+<!-- /TOC -->
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/concepts/global-configuration/users-permissions-roles/_index.md]: let's remove TOC since we might move it back into the layout.

File: content/rancher/v2.x/en/concepts/global-configuration/users-permissions-roles/_index.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+
 
 Within Rancher, each user authenticates as a _user_, which is a login that grants you access to Rancher. As mentioned in [Authenitcation]({{< baseurl >}}/rancher/v2.x/en/concepts/global-configuration/authentication), users can either be local or external.
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/concepts/global-configuration/users-permissions-roles/_index.md]: `Authenitcation` is misspelled

File: content/rancher/v2.x/en/concepts/volumes-and-storage/_index.md [hunk 3]
```diff
@@ -6,16 +6,18 @@ There are two ways to create volumes on Kubernetes: Persistent Volumes (PVs) and
...
 
 ![Persistent Volumes]({{< baseurl >}}/img/rancher/persistent-volume.png)
 
 Rancher allows you to create PVs at cluster level and bind them to PVCs later. Volumes are managed on a per-project basis.
 
 ## Storage Classes
 
-Storage classes allow you to create PVCs dynamically without having to create PVs first. For example, an EBS Storage Class will dynamically create EBS volumes and bind them to PVCs. A storage class is similar to the notion of a “storage driver.” The following figure illustrates how a PVC creation triggers the dynamic provisioning of an underlying EBS volume.
+Storage classes allow you to create PVCs dynamically without having to create PVs first. For example, an Amazon EBS Storage class will dynamically create EBS volumes and bind them to PVCs. A storage class is similar to the notion of a _storage driver_. The following figure illustrates how a PVC creation triggers the dynamic provisioning of an underlying EBS volume.
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/concepts/volumes-and-storage/_index.md]: Class should be capilatized as "Storage Classes" are a k8s concept.

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 4]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+In This Document:
 
-Coming Soon
+<!-- TOC -->
+
+- [Adding a Persistent Volume](#adding-a-persistent-volume)
+- [Adding Storage Classes](#adding-storage-classes)
+- [What's Next?](#whats-next)
+
+<!-- /TOC -->
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: Remove TOC

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 5]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+
+>**Prerequisite:** Completion of all tasks on this page require the `Manage Volumes` [role](../../../concepts/global-configuration/users-permissions-roles/#project-role-reference).
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: I'd rephrase with "In order to work with volumes, the user will require the Manage Volumes role" 

Also, the link is broken for roles

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 6]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+
+## Adding a Persistent Volume
+
+Your containers can store data on themselves, but if a container fails, that data is lost. To solve this issue, Kubernetes offers _persistent volumes_, which are external storage disks or file systems that your containers can access. If a container crashes, its replacement container can access the data in a persistent volume without any data loss.
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: Your containers can store data on themselves, but if a container fails, that data is lost. 
> We shoudl probably reference the fact that all containers are expected to be stateless and by being stateless, if a container fails, the data is lost.

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 7]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+
+1. From the main menu, select **Storage > Persistent Volumes**.
+
+1. Click **Add Volume**.
+
+1. Enter a **Name** for the persistent volume.
+
+1. Select the **Volume Plugin** for the disk type or service that you're using.
+
+  >**Note:** You can only use the `Amazon EBS Disk` volume plugin in an Amazon EKS or Amazon EC2 cluster.
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: 1. Can you use Amazon EBS Disk with Amazon EKS?
2. the note should read that any cloud providers that offer storage will require [cloud provider](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/) to be enabled for the kubernetes cluster. And as an example, it should be that the `Amazon EBS Disk` option will only work with Amazon EC2 clusters with the `aws` cloud provider enabled or Amazon EKS clusters.

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 8]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+         This options sets how many nodes can access the volume, along with the node read/write permissions. The [Kubernetes Documentation](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) includes a table that lists which access modes are supported by the plugins available.
+
+    - [Mount Options](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#mount-options):
+
+         Each volume plugin allows you to specify additional command line options during the mounting process. You can enter these options in the **Mount Option** fields. Consult each plugin's vendor documentation for the mount options available.
+
+    - **Assign to Storage Class:**
+
+         If you later want to automatically provision persistent volumes identical to the volume that you've specified here, assign it a storage class. Later, when you create a workload that includes persistent volume claims, Rancher automatically provisions a persistent volume for each container with a claim.
+
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: As someone who has never worked with volumes, I find this paragraph confusing.

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 9]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+         >**Note:** You must [add a storage class](#adding-storage-classes) before you can assign it to a persistent volume.
+
+1. Click **Save**.
+
+**Result:** Your new persistent volume is created.
+
+For example, this volume is a _hostPath_ volume in Kubernetes:
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: is there supposed to be an example?

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md [hunk 10]
```diff
@@ -3,12 +3,89 @@ title: Adding Storage
...
+1. From the **Global** view, open the cluster for which you want to dynamically provision persistent storage volumes.
+
+1. From the main menu, select `Storage > Storage Classes`. Click `Add Class`.
+
+1. Enter a `Name` for your storage class.
+
+1. From the `Provisioner` drop-down, select the service that you want to use to dynamically provision storage volumes.
+
+1. From the `Parameters` section, fill out the information required for the service to dynamically provision storage volumes. Each provisioner requires different information to dynamically provision storage volumes. Consult the service's documentation for help on how to obtain this information.
+
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: i'd assume the note about persistent volumes and cloud providers are the same for storage classes as well? we should include it here too.

File: content/rancher/v2.x/en/tasks/workloads/deploy-workloads/_index.md [hunk 11]
```diff
@@ -23,11 +23,20 @@ Deploy a workload to run an application in one or more containers.
...
+    - **Health Check**
+    - **Volumes**
     - **Scaling/Upgrade Policy**
 
+        >**Amazon Note for Volumes:**
+        >
+        > To mount an Amazon EBS volume:
+        >
+        >- In [Amazon AWS](https://aws.amazon.com/), the nodes must be in the same Availability Zone and possess IAM permissions to attach/unattach volumes.
+        >- The cluster must be using AWS cloud provider.
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/tasks/workloads/deploy-workloads/_index.md]: link to the k8s cloud provider directions and indicate that the cloud provider can be set in the `cluster options` when creating an AWS EC2 cluster or creating a custom cluster.

**Comment by deniseschannon** [comment]
https://github.com/rancher/docs/issues/374

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md (hunk 10 above)
Comment (MBishop17) [reply content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: which note are you referring to?

File: content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md (hunk 7 above)
Comment (MBishop17) [reply content/rancher/v2.x/en/tasks/clusters/adding-storage/_index.md]: I know the option you're talking about, but I'm still having trouble following this comment. Can you elaborate a little more?
